Products/ZenUI3/browser/resources/js/zenoss/EventPanels.js

```diff
@@ -1490,11 +1490,20 @@
             this.refresh();
         },
         restoreURLState: function() {
-            var qs = window.location.search.replace(/^\?/, ''),
-            state = Ext.urlDecode(qs).state;
-            if (state) {
+            var qs = window.location.search.replace(/^\?/, ''), state, decState;
+            
+            if (qs) {
                 try {
-                    state = Ext.decode(Zenoss.util.base64.decode(decodeURIComponent(state)));
+                    if (Ext.urlDecode(qs).state) {
```

Comment:
`Ext.urlDecode(qs)` is being called 3 times now. It makes more sense to call it once with something like `var decoded = Ext.urlDecode(qs);` *then* check for presence of the `state` field with something like `if(decoded.state){ ... }`.